SPEC.md

```diff
@@ -436,187 +436,37 @@ The Data Connect service responds with the following table:
 
+*   ga4gh_type (described above)
 *   **Logical Operators**
     *   `AND`, `OR`, `NOT`
 *   **Comparison Operators**
     *   `&lt;`, `>`, `&lt;=`, `>=`, `=`, `&lt;>`, `!=`
-    *   `BETWEEN, IS NULL, IS NOT NULL`
-    *   `IS DISTINCT FROM`*
-    *   `IS NOT DISTINCT FROM`*
-    *   `GREATEST`, `LEAST`
-    *   Quantified Comparison Predicates: `ALL`, `ANY` and `SOME`*
+    *   `IS NULL, IS NOT NULL`
     *   Pattern Comparison: `LIKE`
 *   **Conditional Expressions**
-    *   `CASE`, `IF`, `COALESCE`, `NULLIF`
-*   **Conversion Functions**
-    *   `cast(value AS type)` → `type`
-    *   `format(format, args...)` → `varchar`
-*   **Mathematical Functions**
-    *   Most basic functions are supported across implementations. Notably missing are hyperbolic trig functions, infinity, floating point, and statistical/CDF functions.
-    *   `abs(x)` → [same as input]
-    *   `ceil(x)` → [same as input]
-    *   `ceiling(x)` → [same as input]
-    *   `degrees(x)` → `double`*
-    *   `exp(x)` → `double`
-    *   `floor(x)` → [same as input]
-    *   `ln(x)` → `double`
-    *   `log(b, x)` → `double`
-    *   `log10(x)` → `double`
-    *   `mod(n, m)` → [same as input]
-    *   `pi()` → `double`
-    *   `pow(x, p)` → `double`*
-    *   `power(x, p)` → `double`
-    *   `radians(x)` → `double`*
-    *   `round(x)` → [same as input]
-    *   `round(x, d)` → [same as input]
-    *   `sign(x)` → [same as input]
-    *   `sqrt(x)` → `double`
-    *   `truncate(x)` → `double`*
-    *   Random Functions:
-        *   `rand()` → `double`*
-        *   `random()` → `double`*
-        *   `random(n)` → [same as input]*
-        *   `random(m, n)` → [same as input]*
-    *   Trigonometric Functions:
-        *   `acos(x)` → `double`
-        *   `asin(x)` → `double`
-        *   `atan(x)` → `double`
-        *   `atan2(y, x)` → `double`
-        *   `cos(x)` → `double`
-        *   `sin(x)` → `double`
-        *   `tan(x)` → `double`
-*   **Bitwise Functions**
-    *   `bitwise_and(x, y)` → `bigint`
-    *   `bitwise_or(x, y)` → `bigint`
-    *   `bitwise_xor(x, y)` → `bigint`
-    *   `bitwise_not(x)` → `bigint`
-    *   `bitwise_left_shift(value, shift)` → [same as value]
-    *   `bitwise_right_shift(value, shift, digits)` → [same as value]
-    *   `bit_count(x, bits)` → `bigint`*
-*   **Regular Expression Functions**
-    *   `regexp_extract_all(string, pattern)` -> `array(varchar)`*
-    *   `regexp_extract_all(string, pattern, group)` -> `array(varchar)`*
-    *   `regexp_extract(string, pattern)` → `varchar`*
-    *   `regexp_extract(string, pattern, group)` → `varchar`*
-    *   `regexp_like(string, pattern)` → `boolean`*
-    *   `regexp_replace(string, pattern)` → `varchar`*
-    *   `regexp_replace(string, pattern, replacement)` → `varchar`*
-    *   `regexp_replace(string, pattern, function)` → `varchar`*
-*   **UUID Functions**
-    *   `uuid()*`
-*   **Session Information Functions**
-    *   `current_user`*
+    *   `IF`
```

Comment:
If I could only have one conditional expression, I'd want CASE. I can't think of a time I've seen IF in a SQL query, but CASE pops up quite a bit. I'm not aware of any incompatibilities between SQL implementations here.

COALESCE isn't 100% necessary, but it's part of ANSI SQL and because of the way SQL handles nulls, it's often important to have.

Reply:
Replaced IF with CASE.

Reply:
Added COALESCE and IF both actually